### ISSUE TITLE
Allow the pretty-printer to be configured with .eslintrc files

### ISSUE THREAD

**Issue by benjamn** [benjamn/recast#226]
This came up in the discussion over at https://gitter.im/benjamn/recast, thanks to @hzoo.

Though I've known for a long time that the printer isn't very configurable, it seemed like a bad idea to invent a Recast-specific set of configuration options, so I really like the idea of just borrowing the one that `.eslintrc` files already use.

This is potentially a lot of work (adding multiple new cases to each printer rule), but it can be done incrementally, I think.

**Comment by jamestalmage** [comment]
>  This is potentially a lot of work, (adding multiple new cases to each printer rule)

Just to pile on :stuck_out_tongue: , what about `.editorconfig`, and configs to match zero-config linters like `standard`, and `xo`?

I am pretty sure it is possible to coerce all those to `.eslintrc` format (`standard` and `xo` are just `eslint` wrappers, so almost definitely in that case).

**Comment by benjamn** [comment]
I'm happy to pick whichever format can most easily accommodate coerced versions of other formats!

**Comment by juliankrispel** [comment]
Ooooh this would be so neat

**Comment by skovhus** [comment]
What about detecting it based on the existing code? Instead of configuration we can try to detect it by looking at strings in the code (i.e rawValue of string literals).

**Comment by hzoo** [comment]
Yeah that should definetely work for that particular case but it would cover spacing, newlines, etc.
